tools/flaky-test-retryer/github_commenter.go

```diff
@@ -37,3 +53,107 @@ func NewGithubClient(githubAccount string) (*GithubClient, error) {
 	}
 	return &GithubClient{ghc}, err
 }
+
+// PostComment posts a new comment on the PR specified in JobData, retrying the job that triggered it.
+// The comment body is dynamically built based on previous retry comments on this PR, and any old
+// comments are removed before the new one is posted.
+func (gc *GithubClient) PostComment(jd *JobData, outliers []string, dryrun bool) error {
+	oldComment, err := gc.getOldComment(jd.Refs[0].Org, jd.Refs[0].Repo, jd.Refs[0].Pulls[0].Number)
+	if err != nil {
+		return err
+	}
+	oldEntries, err := parseEntries(oldComment)
+	if err != nil {
+		return err
+	}
+	if _, ok := oldEntries[jd.JobName]; !ok {
+		oldEntries[jd.JobName] = 0
+	}
+	newComment, canRetry := buildNewComment(jd, oldEntries, outliers)
+	if !canRetry {
+		return fmt.Errorf("expended all %d retries", maxRetries)
+	}
+	if dryrun {
+		log.Printf("[dry run] Comment not updated. See it here:\n%s\n", newComment)
+		return nil
+	}
+	if oldComment != nil {
+		if err := gc.DeleteComment(jd.Refs[0].Org, jd.Refs[0].Repo, oldComment.GetID()); err != nil {
+			return err
+		}
+	}
+	_, err = gc.CreateComment(jd.Refs[0].Org, jd.Refs[0].Repo, jd.Refs[0].Pulls[0].Number, newComment)
+	return err
+}
+
+// getOldComment queries the GitHub PR specified and gets the comment made by us. If no comment
+// is found, we do not error, since we will be creating a new one anyways.
+func (gc *GithubClient) getOldComment(org, repo string, pull int) (*github.IssueComment, error) {
+	comments, err := gc.ListComments(org, repo, pull)
+	if err != nil {
+		return nil, err
+	}
+	user, err := gc.GetGithubUser()
```

Comment:
I would prefer collect this information up ahead of time, and store it in gc

Reply:
added Login field to gc, populated in the constructor